docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -357,23 +367,107 @@ spec:
 
 * Infrastructure templates are expected to be immutable, so infrastructure template contents do not have to hashed in order to detect
   changes.
-
+   
+##### Remediation (using delete-and-recreate)
+
+- KCP remediation is triggered by the MachineHealthCheck controller marking a machine for remediation. see 
+  [machine-health-checking proposal](https://github.com/kubernetes-sigs/cluster-api/blob/11485f4f817766c444840d8ea7e4e7d1a6b94cc9/docs/proposals/20191030-machine-health-checking.md)
+  for additional details. When there are multiple machines that are marked for remediation, the oldest one will be remediate first.
+  
+- Following rules should be satisfied in order to start remediation
+  - The cluster MUST have at least 3 machines, because this is the smallest cluster size that allows any etcd failure tolerance.
+  - The number of replicas MUST be equal to or greater than the desired replicas. This rule ensures that when the cluster
+    is missing replicas, we skip remediation and instead perform regular scale up/rollout operations first. 
+  - The cluster MUST have no deleting machines. This rule prevents KCP taking actions while the cluster is in a transitional state.
+  - Remediation MUST preserve etcd quorum. This rule ensures that we will not remove a member that would result in etcd
+    losing a majority of members and thus become unable to field new requests.
+
+- When all the conditions for stating remediaiton are satisfied, KCP temporarily suspend any operation in progress
+  in order to perform remediation. 
+- Remediation will be performed by issuing a delete on the unhealthy machine; after deleting the machine, KCP
+  will restore the target number of machines by triggering a scale up (current replicas<desired replicas) and then
+  eventually restart the rollout action.
+  See [why delete and recreate](#why-delete-and-recreate) for an explanation about why KCP should remove the member first and then add its replacement.
+
+###### Why delete and recreate
+
+When replacing a KCP machine the most critical component to be taken into account is etcd, and
+according to the [etcd documentation](https://github.com/etcd-io/etcd/blob/master/Documentation/faq.md#should-i-add-a-member-before-removing-an-unhealthy-member),
+it's important to remove an unhealthy etcd member first and then add its replacement:
+
+_etcd employs distributed consensus based on a quorum model; (n/2)+1 members, a majority, must agree on a proposal before
+it can be committed to the cluster. These proposals include key-value updates and membership changes. This model totally
+avoids any possibility of split brain inconsistency. The downside is permanent quorum loss is catastrophic._
```

Comment:
```suggestion
it can be committed to the cluster. These proposals include key-value updates and membership changes. This model
avoids potential split brain inconsistencies. The downside is permanent quorum loss is catastrophic._
```